lib/pbench/server/api/resources/query_apis/datasets/table_of_content.py

```diff
@@ -0,0 +1,210 @@
+)
+from pbench.server.api.resources.query_apis import CONTEXT
+from pbench.server.api.resources.query_apis.datasets import RunIdBase
+
+
+class TableOfContent(RunIdBase):
+    """
+    Table of Content API returns the list of sub-directories and files
+    present under a directory.
+    """
+
+    MAX_SIZE = 10000
+
+    def __init__(self, config: PbenchServerConfig, logger: Logger):
+        super().__init__(
+            config,
+            logger,
+            Schema(
+                Parameter("run_id", ParamType.STRING, required=True),
+                Parameter("parent", ParamType.STRING, required=True),
+            ),
+        )
+
+    def assemble(self, json_data: JSON, context: CONTEXT) -> JSON:
+        """
+        Construct a pbench Elasticsearch query for getting a list of
+        documents which contains the user provided parent with files
+        and its sub-directories with metadata of run-toc index document
+        that belong to the given run id.
+
+        Args:
+            json_data: JSON dictionary of type-normalized parameters,
+                    we pull 'parent' key value from it.
+            context: propagate the requested set of metadata to other
+                    methods, it includes "run_id" which is Dataset
+                    document ID.
+
+        EXAMPLE:
+        {
+            "run_id": "1234567890"
+            "parent": '/1-default'
+        }
+        """
+        # Copy parent directory metadata to CONTEXT for postprocessor
+        parent = context["parent"] = json_data.get("parent")
+        run_id = context["run_id"]
+        dataset = context["dataset"]
+
+        self.logger.info(
+            "Discover Table of Content run_data_id {}, directory {}",
+            dataset.md5,
+            parent,
+        )
+
+        # Retrieve the ES indices that belong to this run_id from the metadata
+        # table
+
+        indices = self.get_index(dataset, "run-toc")
+        if not indices:
+            self.logger.debug(
+                f"Found no indices matching the prefix run-toc"
+                f" for a dataset {dataset!r}"
+            )
+            abort(HTTPStatus.NOT_FOUND, message="Found no matching indices")
+
+        return {
+            "path": f"/{indices}/_search",
+            "kwargs": {
+                "json": {
+                    "size": self.MAX_SIZE,
+                    "query": {
+                        "bool": {
+                            "filter": [
+                                {
+                                    "dis_max": {
+                                        "queries": [
+                                            {"match_phrase": {"directory": parent}},
+                                            {"match_phrase": {"parent": parent}},
+                                        ]
+                                    }
+                                },
+                                {"match_phrase": {"run_data_parent": run_id}},
```

Comment:
Here, also; `"run_data_parent"` is a keyword field, so `"match_phrase"` seems at best more processing than we want?

Reply:
Per my [previous comment](https://github.com/distributed-system-analysis/pbench/pull/2507#discussion_r788921236):  I expect that we want to use `"term"` rather than `"match_phrase"`, here.